src/Http/Http/src/StreamPipeReader.cs

```diff
@@ -60,9 +60,9 @@ public StreamPipeReader(Stream readingStream, StreamPipeReaderOptions options)
                 throw new ArgumentOutOfRangeException(nameof(options.MinimumReadThreshold));
             }
 
-            _minimumSegmentSize = options.MinimumSegmentSize;
             _minimumReadThreshold = Math.Min(options.MinimumReadThreshold, options.MinimumSegmentSize);
-            _pool = options.MemoryPool;
+            _pool = options.MemoryPool == MemoryPool<byte>.Shared ? null : options.MemoryPool;
+            _bufferSize = _pool == null ? options.MinimumSegmentSize : Math.Min(options.MinimumSegmentSize, _pool.MaxBufferSize);
```

Comment:
Rename public property too?

Reply:
Should we throw in the case where MinimumSegmentSize is larger then MaxBufferSize?

Reply:
Yep, I plan to make all of the corefx changes in here in another PR. I think these types need a bit more bake time in this repo first.